torch/nn/attention/flex_attention.py

```diff
@@ -447,33 +486,20 @@ def _create_sparse_block_from_block_mask(
 ) -> BlockMask:
     full_blocks, partial_blocks = block_mask
 
-    def create_sparse_block_from_block_mask_inner(block_mask) -> Tuple:
-        block_mask = block_mask.to(dtype=torch.int32)
-        kv_num_blocks = block_mask.sum(dim=3)
-        kv_indices = torch.argsort(block_mask, dim=3, descending=True, stable=True)
-        q_num_blocks = block_mask.sum(dim=2)
-        q_indices = torch.argsort(
-            block_mask, dim=2, descending=True, stable=True
-        ).permute(0, 1, 3, 2)
-        return (
-            kv_num_blocks.to(torch.int32).to(block_mask.device).contiguous(),
-            kv_indices.to(torch.int32).to(block_mask.device).contiguous(),
-            q_num_blocks.to(torch.int32).to(block_mask.device).contiguous(),
-            q_indices.to(torch.int32).to(block_mask.device).contiguous(),
-        )
-
-    full_bm = create_sparse_block_from_block_mask_inner(full_blocks)
+    full_bm = _dense_to_ordered(full_blocks)
     if partial_blocks is not None:
-        partial_bm = create_sparse_block_from_block_mask_inner(partial_blocks)
+        partial_bm = _dense_to_ordered(partial_blocks)
     else:
-        partial_bm = (None, None, None, None)
+        partial_bm = (None, None)
 
     return BlockMask(  # type: ignore[call-arg]
-        *full_bm,
-        *partial_bm,
-        KV_BLOCK_SIZE,
-        Q_BLOCK_SIZE,
-        mask_fn,
+        full_bm[0],
+        full_bm[1],
```

Comment:
why not splat?

Reply:
It's smaller now and causes a bunch of mypy issues lol.